redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewBox.kt

```diff
@@ -36,66 +34,91 @@ import app.cash.redwood.layout.widget.Box
+
+  private val density = Density(context.resources)
 
   private var defaultHorizontalAlignment = CrossAxisAlignment.Start
   private var defaultVerticalAlignment = CrossAxisAlignment.Start
+  private var width = Constraint.Fill
+  private var height = Constraint.Fill
 
-  override val value = object : FrameLayout(context) {
-    init {
-      layoutParams = LayoutParams(MATCH_PARENT, MATCH_PARENT)
+  init {
+    layoutParams = LayoutParams(MATCH_PARENT, MATCH_PARENT)
+  }
 
-      // TODO: Remove this.
-      setBackgroundColor(0xFFFFFF66.toInt())
-    }
+  override fun width(width: Constraint) {
+    this.width = width
+  }
 
-    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
-      val widgetChildren = this@ViewBox.children.widgets
+  override fun height(height: Constraint) {
+    this.height = height
+  }
 
-      for (child in widgetChildren) {
-        child.value.layoutParams = toLayoutParams(child.modifier)
-        child.value.requestLayout()
+  override fun margin(margin: Margin) {
+    updateLayoutParams {
+      val layoutParams = this as MarginLayoutParams
+      with(density) {
+        layoutParams.setMargins(
+          margin.start.toPxInt(),
+          margin.top.toPxInt(),
+          margin.end.toPxInt(),
+          margin.bottom.toPxInt(),
+        )
       }
+    }
+  }
 
-      // Get width and height values from layout parameters
-      val width = when (layoutParams.width) {
-        LayoutParams.MATCH_PARENT -> widthMeasureSpec
-        LayoutParams.WRAP_CONTENT -> widgetChildren.maxOfOrNull { it.value.measuredWidth } ?: 0
-        else -> layoutParams.width
-      }
+  override fun horizontalAlignment(horizontalAlignment: CrossAxisAlignment) {
+    this.defaultHorizontalAlignment = horizontalAlignment
+  }
 
-      val height = when (layoutParams.height) {
-        LayoutParams.MATCH_PARENT -> heightMeasureSpec
-        LayoutParams.WRAP_CONTENT -> widgetChildren.maxOfOrNull { it.value.measuredHeight } ?: 0
-        else -> layoutParams.height
-      }
+  override fun verticalAlignment(verticalAlignment: CrossAxisAlignment) {
+    this.defaultVerticalAlignment = verticalAlignment
+  }
 
-      // Call to super here triggers a layout pass on the children.
-      super.onMeasure(widthMeasureSpec, heightMeasureSpec)
-      setMeasuredDimension(width, height)
+  /** Flush Redwood's modifiers into FrameLayout's LayoutParams before it measures. */
+  override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+    var widthMode = MeasureSpec.getMode(widthMeasureSpec)
+    val widthSize = MeasureSpec.getSize(widthMeasureSpec)
+    var heightMode = MeasureSpec.getMode(heightMeasureSpec)
+    val heightSize = MeasureSpec.getSize(heightMeasureSpec)
+
+    for (child in children.widgets) {
+      val layoutParams = child.value.layoutParams as LayoutParams
+      layoutParams.setFrom(child.modifier)
+      child.value.layoutParams = layoutParams // To force layout.
```

Comment:
```suggestion
      child.value.updateLayoutParams<LayoutParams> {
        setFrom(child.modifier)
      }
```
https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:core/core-ktx/src/main/java/androidx/core/view/View.kt;l=326-332?q=updateLayoutParams&start=11